airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -350,6 +353,19 @@ def _dynamic_stream_configs(
                 if "type" not in dynamic_stream:
                     dynamic_stream["type"] = "DeclarativeStream"
 
+                # Ensure that each stream is created with a unique name
+                name = dynamic_stream.get("name")
+
+                if name in seen_dynamic_streams:
+                    error_message = f"Dynamic streams list contains a duplicate name: {name}. Please check your configuration."
+
+                    raise AirbyteTracedException(
+                        message=error_message,
+                        internal_message=error_message,
+                        failure_type=FailureType.config_error,
```

Comment:
I'm wondering if we need to do the distinction between dynamic streams from config and dynamic streams from API. If this error comes for the config resolver, than it is a config error. If this error comes from the API, it's probably a system error. WDYT?

Reply:
It depends on how we think about the manifest. Is this a configuration for the source as well? If so, this could be treated as a configuration error because the problem lies in the endpoint defined in the manifest. If not, it should depend on the type of resolver.

I think it's okay to categorize it as a configuration error because, in my view, the description of the manifest involves configuring the source.

Reply:
I see it the other way. A config issue is an issue that the Airbyte user can fix by updating the config. An Airbyte user won't easily have access to update the description of the manifest so in this case, it should be flagged as a system error. Does that make sense?

Reply:
Ok. I have added condition for failure_type and unit test for it.